tests/unit/test_encrypt.py

```diff
@@ -49,3 +92,122 @@ def _padding_cycle(self, data: ContestData) -> None:
+        nonce = rand_q()
+        encryption_seed = ONE_MOD_Q
+        contest_description = get_sample_contest_description()
+        contest = contest_from(contest_description)
+        contest_hash = contest_description.crypto_hash()
+
+        # Add Overvotes
+        for selection in contest.ballot_selections:
+            selection.vote = 1
+
+        # Act
+        encrypted_contest = encrypt_contest(
+            contest,
+            contest_description,
+            keypair.public_key,
+            encryption_seed,
+            nonce,
+            should_verify_proofs=True,
+        )
+
+        # Assert
+        self.assertIsNotNone(encrypted_contest)
+        self.assertIsNotNone(encrypted_contest.extended_data)
+        self.assertTrue(
+            encrypted_contest.is_valid_encryption(
+                contest_hash, keypair.public_key, encryption_seed
+            )
+        )
+
+        # Act
+        decrypted_data = get_optional(
+            encrypted_contest.extended_data.decrypt(keypair.secret_key, encryption_seed)
+        )
+        contest_data = ContestData.from_bytes(decrypted_data)
+
+        # Assert
+        self.assertIsNotNone(contest_data)
+        self.assertIsNotNone(contest_data.error)
+        self.assertIsNotNone(contest_data.error_data)
+        self.assertEqual(contest_data.error, ContestErrorType.OverVote)
+        self.assertGreater(len(contest_data.error_data), 0)
+
+    def test_contest_encrypt_with_write_ins(self):
+
+        # Arrange
+        keypair = get_optional(elgamal_keypair_from_secret(TWO_MOD_Q))
+        nonce = rand_q()
+        encryption_seed = ONE_MOD_Q
+        contest_description = get_sample_contest_description()
+        contest = contest_from(contest_description)
+        contest_hash = contest_description.crypto_hash()
+
+        # Add Write-ins
+        for selection in contest.ballot_selections:
+            selection.write_in = "write_in"
+
+        # Act
+        encrypted_contest = encrypt_contest(
+            contest,
+            contest_description,
+            keypair.public_key,
+            encryption_seed,
+            nonce,
+            should_verify_proofs=True,
+        )
+
+        # Assert
+        self.assertIsNotNone(encrypted_contest)
+        self.assertIsNotNone(encrypted_contest.extended_data)
+        self.assertTrue(
+            encrypted_contest.is_valid_encryption(
+                contest_hash, keypair.public_key, encryption_seed
+            )
+        )
+
+        # Act
+        decrypted_data = get_optional(
+            encrypted_contest.extended_data.decrypt(keypair.secret_key, encryption_seed)
+        )
+        contest_data = ContestData.from_bytes(decrypted_data)
+
+        # Assert
+        self.assertIsNotNone(contest_data)
+        self.assertIsNotNone(contest_data.write_ins)
+        if contest_data is not None and contest_data.write_ins is not None:
+            self.assertGreater(len(contest_data.write_ins), 0)
```

Comment:
Why not `for selection in contest_data: self.assertEqual("write_in", selection.write_in)` to match the set up on line 182?

Reply:
Can add.